test/Test/Chapter3.hs

```diff
@@ -11,3 +11,52 @@ chapter3 :: Spec
 chapter3 = describe "Chapter3" $ do
     describe "Chapter3Normal" $ it "" $ True `shouldBe` True
     describe "Chapter3Advanced" $ it "" $ True `shouldBe` True
+    describe "Task2: fight beetween knight and moster" $ do
```

Comment:
Amazing job on writing tests :gem: I see, that you've spent a lot of time on tests... This is really nice!